viadot/tasks/cloud_for_customers.py

```diff
@@ -128,12 +154,33 @@ def run(
             fields (List[str], optional): The C4C Table fields. Defaults to None.
             params (Dict[str, Any]): The query parameters like filter by creation date time. Defaults to json format.
             if_empty (str, optional): What to do if query returns no data. Defaults to "warn".
+            credentials_secret (str, optional): The name of the Azure Key Vault secret containing a dictionary
+            with C4C credentials. Defaults to None.
+            vault_name (str, optional): The name of the vault from which to obtain the secret. Defaults to None.
 
         Returns:
             pd.DataFrame: The query result as a pandas DataFrame.
         """
+        if not credentials_secret:
+            # attempt to read a default for the service principal secret name
+            try:
+                credentials_secret = PrefectSecret("C4C_AIA").run()
+            except ValueError:
+                pass
+
+        if credentials_secret:
+            credentials_str = AzureKeyVaultSecret(
+                credentials_secret, vault_name=vault_name
+            ).run()
+            credentials = json.loads(credentials_str)[env]
+
         cloud_for_customers = CloudForCustomers(
-            url=url, params=params, endpoint=endpoint, env=env, fields=fields
+            url=url,
+            params=params,
+            endpoint=endpoint,
+            env=env,
+            fields=fields,
+            credentials=credentials,
```

Comment:
If credentials are None (for instance: test_cloud_for_customers:test_c4c_to_df()) this will fail 😭

Reply:
When I run the tests, test_c4c_to_df() fails with an error:
```UnboundLocalError: local variable 'credentials' referenced before assignment```.

The test uses `C4CToDF` class and its run() method, without a `credentials_secret` parameter. This should be handled - right now, in run() method, CloudForCustomers() is being called on line 177, with the `credentials` parameter. Take a closer look on when the credentials parameter is being defined ✨ 😱